indexing-service/src/main/java/org/apache/druid/indexing/overlord/ForkingTaskRunner.java

```diff
@@ -219,6 +220,23 @@ public TaskStatus call()
                           );
                         }
 
+                        // Override task specific javaOptsArray
+                        try {
+                          List<String> taskJavaOptsArray = jsonMapper.convertValue(
+                              task.getContextValue(ForkingTaskRunnerConfig.JAVA_OPTS_ARRAY_PROPERTY),
+                              new TypeReference<List<String>>() {}
+                          );
+                          if (taskJavaOptsArray != null) {
+                            Iterables.addAll(command, taskJavaOptsArray);
+                          }
+                        }
+                        catch (Exception e) {
+                          throw new IllegalArgumentException(
+                              "javaOptsArray in context of task: " + task + " must be an array of strings.",
```

Comment:
we should replace `javaOptsArray` with `ForkingTaskRunnerConfig.JAVA_OPTS_ARRAY_PROPERTY`.
Also, I think task doesn't implement a `toString` properly generally - should be replaced with `taskId`

Reply:
done